### PR TITLE
Updating npm packages dependencies

### PR DESCRIPTION
**Pull request by jesperronn**
I put this in WIP as I expect to see some errors until all updates are in place.

Eventually, this will update all npm dependencies to most recent versions.

* [ ] I am NOT updating `karma-jasmine` here, since getting up to Jasmine2 will require more changes.
       Feel free to open a new pull-request.

I updated
* [x] chai
* [x] colors
* [x] grunt-concurrent
* [x] grunt-contrib-*
* [x] karma
* [x] karma-*-launcher
* [x] karma-coverage
* [x] mocha
* [x] semver
* [x] yiewd

### CODE REVIEW

File: bower.json [hunk 1]
```diff
@@ -3,6 +3,7 @@
   "version": "0.11.1",
   "main": "dist/typeahead.bundle.js",
   "dependencies": {
+    "jasmine-jquery": "~2.1.1",
```
Comment (jesperronn) [inline bower.json]: I believe this is an error, since jasmine-jquery is put in devDependencies section.

Will fix

File: bower.json [hunk 2]
```diff
@@ -8,6 +8,6 @@
   "devDependencies": {
     "jquery": "~1.7",
     "jasmine-ajax": "~1.3.1",
-    "jasmine-jquery": "~1.5.2"
+    "jasmine-jquery": "~2.1.1"
```
Comment (jesperronn) [inline bower.json]: I am uncertain if this has to be rolled back to version 1.7.0 or perhaps a version 2 before they added the dependency to jquery 2 in jasmine-jquery library.

See discussion https://github.com/velesin/jasmine-jquery/issues/267

**Comment by CoreSystemDevelopment** [comment]
Merge?

**Comment by jesperronn** [comment]
@CoreSystemDevelopment this branch works. I will clean up and squash the changes so that it's ready to merge.

**Pull request merged**